conans/client/conan_api.py

```diff
@@ -644,29 +644,21 @@ def copy(self, reference, user_channel, force=False, packages=None):
                  self._user_io, self._remote_manager, self._registry, force=force)
 
     @api_method
-    def authenticate(self, name, password, remote=None):
-        if not remote:
-            remote = self._registry.default_remote
-        else:
-            remote = self._registry.remote(remote)
-        if password == "":
-            name, password = self._user_io.request_login(remote_name=remote.name,
-                                                         username=name)
-        self._remote_manager.authenticate(remote, name, password)
+    def authenticate(self, name, password, remote):
+        _, remote, prev_user, user = self._remote_manager.authenticate(remote, name, password)
+        return remote, prev_user, user
 
     @api_method
     def user_set(self, user, remote_name=None):
```

Comment:
Here "remote_name".  Should we rename all to "remote" when it is a "remote_name"? Are there more "remotes" being complex objects or not?